File: submissions/cpp/100.cpp
class Solution {
public:
    bool isSameTree(TreeNode *p, TreeNode *q) {
        if (!p && !q)
            return true;
        else if (p && q) {
            if (p->val == q->val)
                return isSameTree(p->left, q->left) && isSameTree(p->right, q->right);
            return false;
        } else
            return false;
    }
};

File: LeetCode/python3/100.py
class Solution:
    def isSameTree(self, p: 'TreeNode', q: 'TreeNode') -> 'bool':
        if not p and not q:
            return True
        elif not p or not q:
            return False
        return q.val == p.val and self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)